### ISSUE TITLE
`aria-required` not applied on some input components when they are required fields

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#3896]
### Summary of the bug 
When applying the `required` prop on some input components, the `aria-required` attribute doesn't get programmatically applied.

On these components, the word required can still be heard by a screen reader user (e.g. by appending the word "required" onto the aria-label), but `aria-required` should be programmatically applied as well to ensure it will always get read out, and it will always be clear that the field is required.

### 🪜 How to reproduce 
1. Go the [Required single select example](https://design.sis.gov.uk/components/select/code#required).
2. Inspect the HTML.
3. See that "required" is appended onto the aria-label on the `<button>`, but `aria-required` is not applied
<img width="335" height="109" alt="Image" src="https://github.com/user-attachments/assets/fb740f79-ddf5-47a3-8ba5-4eb43cd0ac3e" />

### 🧐 Expected behaviour 
`aria-required` should be applied when an input component is set as required.

I think these are the components / elements which need `aria-required`:
- The `<input>`s in the date input / date picker (could maybe use [`required` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/required) here instead)
- The `<input>`s in the time input (could also maybe use `required`)
- The `<fieldset>` in checkbox groups
- The `<div>` with `role="radiogroup"` on radio button groups
- The `<button>` on a select (single-select)

### 🚨 Urgency (low, medium or high)
High urgency - the team this ticket was raised on behalf need it to be completed as soon as possible. Also it's an accessibility issue.

### Additional info 
This was raised on behalf of a customer. This issue was pointed out by a tester their team has been working with.

**Comment by GCHQ-Developer-847** [comment]
The customer I raised this ticket for is using the `ic-select` and `ic-radio-group` components. I have included in the description all other components which I think need changes. If the changes end up being quite complex / take a long time, maybe just do these two components and create a ticket for the others (as the customer needs this resolved ASAP).

**Comment by GCHQ-Developer-847** [comment]
As the word "required" is included in aria-labels / `aria-labelledby` for these components, some screen reader users may hear the word "required" twice when adding `aria-required`. However, I think this is ok?

Maybe it would be good to explore whether the word "required" can be removed from the aria-labels to avoid this duplication. But I don't think it's necessary. I think there was a good reason for not doing it this way in the first place (e.g. see [date input comment](https://github.com/mi6/ic-ui-kit/blob/e673b5f5f6eb7aaeac27f2dea1ca5351aab8e86f/packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx#L1146-L1147)).
